modules/apps_api/lib/apps_api/notification_service.rb

```diff
@@ -65,16 +68,20 @@ def create_hash(app_record:, user:, published:)
     end
 
     def event_is_invalid(event)
-      event['outcome']['result'] != 'SUCCESS' ||
+      # checking if the event is unable to be processed,
+      # or has already been processed.
+      @handled_events.include?(event['uuid']) ||
```

Comment:
my gut says that extracting compound conditionals here would make this logic easier to parse. What do you think?

Reply:
➕

Reply:
extracted to separate methods in latest commit